### PR TITLE
[Deploy] MMYOLO model convert to onnx for deployment.

### PR DESCRIPTION
**Pull request by triple-Mu**
In order to make the deployment process simpler and more user-friendly. 
This pr will remove the dependency of mmdeploy. and only use pytorch and onnx's native api to export models to onnx for users to continue to convert.

### CODE REVIEW

File: mmyolo/easydeploy/backbone/focus.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        self.__dict__.update(orin_Focus.__dict__)
+
+    def forward(self, x: Tensor) -> Tensor:
+        B, C, H, W = x.shape
+        x = x.reshape(B, C, -1, 2, W)
+        x = x.reshape(B, C, x.shape[2], 2, -1, 2)
+        half_H = x.shape[2]
+        half_W = x.shape[4]
+        x = x.permute(0, 5, 3, 1, 2, 4)
+        x = x.reshape(B, C * 4, half_H, half_W)
```
Comment (PeterH0323) [inline mmyolo/easydeploy/backbone/focus.py:21]:
```suggestion
        batch_size, channel, height, width = x.shape
        x = x.reshape(batch_size, channel, -1, 2, width)
        x = x.reshape(batch_size, channel, x.shape[2], 2, -1, 2)
        half_h = x.shape[2]
        half_w = x.shape[4]
        x = x.permute(0, 5, 3, 1, 2, 4)
        x = x.reshape(B, C * 4, half_h, half_w)
```

File: mmyolo/easydeploy/backbone/focus.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+class NcnnFocus(nn.Module):
+
+    def __init__(self, orin_Focus: nn.Module):
+        super().__init__()
+        self.__dict__.update(orin_Focus.__dict__)
+
+    def forward(self, x: Tensor) -> Tensor:
+        batch_size, c, h, w = x.shape
```
Comment (PeterH0323) [inline mmyolo/easydeploy/backbone/focus.py:33]: 建议全部单个字母的变量都扩展成单词，这样用户容易懂一些

**Comment by hhaAndroid** [comment]
@triple-Mu Please refer to https://github.com/open-mmlab/mmocr/pull/1524 This PR is used to reorganize the file structure

**Comment by triple-Mu** [comment]
> @triple-Mu Please refer to [open-mmlab/mmocr#1524](https://github.com/open-mmlab/mmocr/pull/1524) This PR is used to reorganize the file structure

After the project is moved to the project dir, is it a python package or a normal script file?

**Comment by hhaAndroid** [comment]
@triple-Mu 
- 所有文件都应该在 project 路径下
- pypi 中的包不会包括这个部分
- 使用使用的时候，可以自己导入或者通过 custom_imports 触发装饰器

File: projects/easydeploy/Wrapper.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
+import warnings
```
Comment (PeterH0323) [inline projects/easydeploy/Wrapper.py:1]: 文件名要全小写 `Warpper.py` -> `warpper.py`

**Comment by hhaAndroid** [comment]
@triple-Mu

File: docs/zh_cn/deploy/model_convert.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# MMYOLO 模型转换
```
Comment (hhaAndroid) [inline docs/zh_cn/deploy/model_convert.md:1]: 对于 easydeploy 的任何东西都应该放到 project下，包括 md。

到时候合入后，我们会在 readme 主页提供入口跳转的。

File: projects/easydeploy/backbone/__init__.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) OpenMMLab. All rights reserved.
```
Comment (hhaAndroid) [inline projects/easydeploy/backbone/__init__.py:1]: easydeploy 底下需要一个 readme 入口文档。docs 文件夹是否需要看你自己，但是对外入口 readme 是必须要有的哦

File: projects/easydeploy/README_zh-CN.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+## 介绍
+
+本项目作为 MMYOLO 的部署 project 单独存在，意图剥离 MMDeploy 当前的体系，独自支持用户完成模型训练后的转换和部署功能，使用户的学习和工程成本下降。
+
+当前支持对 ONNX 格式和 TensorRT 格式的转换，后续对其他推理平台也会支持起来。
+
+## 转换教程
+
+- [To onnx](docs/model_convert.md)
```
Comment (hhaAndroid) [inline projects/easydeploy/README_zh-CN.md:11]:
```suggestion
- [Model Convert](docs/model_convert.md)
```

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**